### PR TITLE
TCM-410 | Consolidate Authentication

### PR DESCRIPTION
**Pull request by vespertineSiren**
### Summary

- Handled all the major issues on the ticket. 

Note: Ashley's account will crash the app because of GSON errors. Something to consider moving forward. Onboarding ticket follow up since theres a scrollview being used in onboarding [ticket](https://hyfnla.atlassian.net/browse/TCM-422)

### Issue

[TCM-410](https://hyfnla.atlassian.net/browse/TCM-410)

### Video/Screenshot

**First time login will show Onboarding. Logging out and registering a new account on the device will not show onboarding**

https://user-images.githubusercontent.com/27742105/120051485-aad5a700-bfd5-11eb-985f-9a0885b6a614.mov

**Registering for the first time will show the Onboarding. Logging out and logging in with a new account will not show the onboarding.**

https://user-images.githubusercontent.com/27742105/120051692-77dfe300-bfd6-11eb-93e9-54f4e2c207f3.mov

### CODE REVIEW

File: app/src/main/java/com/trx/consumer/managers/BackendManager.kt [hunk 1]
```diff
@@ -224,7 +224,15 @@ class BackendManager(private val api: BaseApi, private val cacheManager: CacheMa
...
-        return call(RequestModel(endpoint = EndpointModel.UPDATE, path = path, params = params))
+        val response = call(
+            RequestModel(
+                endpoint = EndpointModel.UPDATE,
+                path = path,
+                params = params
+            )
+        )
+        if (response.isSuccess) user()
+        return response
```
Comment (onodude) [inline app/src/main/java/com/trx/consumer/managers/BackendManager.kt:235]: might be a good idea to do like this with two returns
```
if (!response.isSuccess) return response
return user()
```

File: app/src/main/java/com/trx/consumer/managers/CacheManager.kt [hunk 2]
```diff
@@ -81,4 +83,36 @@ class CacheManager(context: Context) {
...
+            dataStore.edit {
+                it[kDidShowOnboarding] = value
+            }
+        }
+    }
+
+    suspend fun didShowRestore(): Boolean {
+        return withContext(Dispatchers.IO) {
+            dataStore.data.map {
+                it[kDidShowRestore]
```
Comment (onodude) [inline app/src/main/java/com/trx/consumer/managers/CacheManager.kt:106]: if isVersion1Enabled return true

**Review by onodude — CHANGES_REQUESTED** [review]
thank you @vespertineSiren added a couple of comments

File: app/src/main/java/com/trx/consumer/managers/BackendManager.kt (hunk 1 above)
Comment (vespertineSiren) [reply app/src/main/java/com/trx/consumer/managers/BackendManager.kt:235]: did this instead `return if (!response.isSuccess) response else user()`

Comment (vespertineSiren) [reply app/src/main/java/com/trx/consumer/managers/BackendManager.kt]: Here we go https://github.com/hyfn/trx-consumer-android/pull/82/commits/4e17dfad34349b3070e75c12567a128c77f520af

File: app/src/main/java/com/trx/consumer/managers/CacheManager.kt (hunk 2 above)
Comment (vespertineSiren) [reply app/src/main/java/com/trx/consumer/managers/CacheManager.kt]: Here we go: https://github.com/hyfn/trx-consumer-android/pull/82/commits/dfde79cee102d681f80442556cb0aa514bf05b23

File: app/src/main/java/com/trx/consumer/managers/BackendManager.kt (hunk 1 above)
Comment (onodude) [reply app/src/main/java/com/trx/consumer/managers/BackendManager.kt]: haha not sure i suggested the negation but yeah will keep for now

**Review by onodude — APPROVED** [review]
thank you!

**Pull request merged**